test/unit/active_model/serializer/associations_test.rb

```diff
@@ -14,6 +14,36 @@ def test_associations_inheritance
         assert_equal([:comments],
                      another_inherited_serializer_klass._associations.keys)
       end
+      def test_multiple_nested_associations
+        parent = SelfReferencingUserParent.new(name: "The Parent")
```

Comment:
You should use the model attributes to define the `type` instead of defining it directly within the model e.g.:
```ruby
class SelfReferencingUser < Model
end
```
```ruby
parent = SelfReferencingUserParent.new(name: "The Parent", type: Type.new(name:"N1"))
child1 = SelfReferencingUser.new(name: "The child", parent: parent, type: Type.new(name:"N2"))
```